### PR TITLE
Add unit tests for db/repos/transactions* - Closes #1594

### LINKED ISSUE #1594: Add unit test coverage for db/repos/transactions.*
Parent: #1484

### Expected behavior
We need 100% coverage for different type of transactions db/repos/transactions.*

### Actual behavior
Currently there is no coverage for this repo.

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?
No unit test coverage for `db/repos/transactions*`

### How did I fix it?
Added test cases thoroughly for every method in `db/repos/transactions*`. And got 100% coverage for that file.

### How to test it?
```
npx mocha test/unit/db/repos/transactions.*
```
### Review checklist

* The PR solves #1594
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: test/unit/db/repos/transactions/in_transfer.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+				expect(inTransferRepo.dbFields).to.be.eql(['dappId', 'transactionId']);
+
+				expect(inTransferRepo.cs).to.be.an('object');
+				expect(inTransferRepo.cs).to.not.empty;
+				return expect(inTransferRepo.cs).to.have.all.keys('insert');
+			});
+		});
+
+		describe('save()', () => {
+			it('should insert entry into "delegates" table for type 6 transactions', function*() {
```
Comment (jondubois) [inline test/unit/db/repos/transactions/in_transfer.js]: I don't understand how this is related to 'delegates'. Is it a typo?

File: test/unit/db/repos/transactions/out_transfer.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+				]);
+
+				expect(outTransferRepo.cs).to.be.an('object');
+				expect(outTransferRepo.cs).to.not.empty;
+				return expect(outTransferRepo.cs).to.have.all.keys('insert');
+			});
+		});
+
+		describe('save()', () => {
+			it('should insert entry into "delegates" table for type 7 transactions', function*() {
```
Comment (jondubois) [inline test/unit/db/repos/transactions/out_transfer.js]: Is the word 'delegate' correct here?

File: test/unit/db/repos/transactions/transfer.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+				expect(transferRepo.dbFields).to.be.eql(['data', 'transactionId']);
+
+				expect(transferRepo.cs).to.be.an('object');
+				expect(transferRepo.cs).to.not.empty;
+				return expect(transferRepo.cs).to.have.all.keys('insert');
+			});
+		});
+
+		describe('save()', () => {
+			it('should insert entry into "delegates" table for type 0 transactions', function*() {
```
Comment (jondubois) [inline test/unit/db/repos/transactions/transfer.js]: Again 'delegates'?

File: test/unit/db/repos/transactions/vote.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+				expect(voteRepo.dbFields).to.be.eql(['votes', 'transactionId']);
+
+				expect(voteRepo.cs).to.be.an('object');
+				expect(voteRepo.cs).to.not.empty;
+				return expect(voteRepo.cs).to.have.all.keys('insert');
+			});
+		});
+
+		describe('save()', () => {
+			it('should insert entry into "delegates" table for type 3 transactions', function*() {
```
Comment (jondubois) [inline test/unit/db/repos/transactions/vote.js]: Again 'delegates'?

File: test/unit/db/repos/transactions/multisignature.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+				]);
+
+				expect(multiSignaturesRepo.cs).to.be.an('object');
+				expect(multiSignaturesRepo.cs).to.not.empty;
+				return expect(multiSignaturesRepo.cs).to.have.all.keys('insert');
+			});
+		});
+
+		describe('save()', () => {
+			it('should insert entry into "delegates" table for type 4 transactions', function*() {
```
Comment (jondubois) [inline test/unit/db/repos/transactions/multisignature.js]: Word 'delegates' is correct to refer to the affected table name?

File: test/unit/db/repos/delegates.js [hunk 6]
```diff
@@ -165,10 +167,50 @@ describe('db', () => {
...
+				});
+				yield db.transactions.save(trs1);
+				yield db.query(
+					db.$config.pgp.helpers.insert(
+						{
+							transactionId: trs1.id,
+							username: randomstring.generate({
+								length: 10,
+								charset: 'alphabetic',
+							}),
```
Comment (jondubois) [inline test/unit/db/repos/delegates.js]: This kind of randomness is probably not too dangerous (at least length is fixed) but does it doesn't bring any value - If we get an error, how would we reproduce; if it fails randomly, we won't know which specific string caused the error.

File: test/unit/db/repos/delegates.js [hunk 7]
```diff
@@ -14,8 +14,10 @@
 
 'use strict';
 
+const randomstring = require('randomstring');
```
[the remark this answers is not in the document]
Comment (nazarhussain) [reply test/unit/db/repos/delegates.js]: @jondubois We are using `randomstring` across the test suit. I don't think that's something brittle if we domain the specific input domain of the value and randomize in it. e.g. generating transaction ids, block ids etc

File: test/unit/db/repos/delegates.js (hunk 6 above)
Comment (nazarhussain) [reply test/unit/db/repos/delegates.js]: @jondubois As per my understandings randomizing input gave you variety of input on each run. As we run tests many times to finally reached to release, so this way we can test different variety of input values. And its not here, we are using this approach across all test suit.

**Comment by nazarhussain** [comment]
@diego-G Naming for unit test files followed as exactly the same file its testing with same file structure. For functional tests if you group/identify names with functionality that's completely other thing not related to naming convention.

**Pull request merged**